managed/src/main/java/com/yugabyte/yw/controllers/CloudProviderController.java

```diff
@@ -134,6 +141,14 @@ public Result delete(UUID customerUUID, UUID providerUUID) {
         accessKey.delete();
       }
       NodeInstance.deleteByProvider(providerUUID);
+
+      int providersCount = Provider.getbyCode(provider.code).size();
+      // Delete Intance type only if 1 provider with code.
+      // More than 1 provider can have shared instance type so we can't delete it.
+      // as instance type are shred acrossed providers
+      if (providersCount == 1) {
```

Comment:
I like this code but I don't like the comment - it is written slightly casually (a lot of typos/errors). Please remember that this code is open source so the text could be read by other people.

// Delete Instance type only if 1 provider with such code exists.
// More than 1 provider can have this instance type so we can't delete it
// if it is shared across providers.

(Something like this, but it is not a standard. Check comments carefully, please.)

Reply:
Updated comment let me know if looks fine now.